diff --git a/fdroidserver/update.py b/fdroidserver/update.py
index 4c21718003812ad9f6e42e992fbfa423f41b25d6..20a9088c7bb8ee316ec64d354f739215e34a4b6e 100644
--- a/fdroidserver/update.py
+++ b/fdroidserver/update.py
@@ -1500,8 +1500,8 @@ def process_apk(apkcache, apkfilename, repodir, knownapks, use_date_from_apk=Fal
         # resize existing icons for densities missing in the APK
         fill_missing_icon_densities(empty_densities, iconfilename, apk, repodir)
 
-        if use_date_from_apk and manifest.date_time[1] != 0:
-            default_date_param = datetime(*manifest.date_time)
+        if use_date_from_apk:
+            default_date_param = datetime.fromtimestamp(os.stat(apkfile).st_ctime)
         else:
             default_date_param = None
 
